internal/sftp/goclient.go

```diff
@@ -57,6 +57,26 @@ func (c *GoClient) Upload(ctx context.Context, src io.Reader, dest string) (int6
 	return bytes, nil
 }
 
+// Delete removes the data from dest, a new SFTP connection is opened before
+// removing the file, and closed when the delete is complete.
+func (c *GoClient) Delete(ctx context.Context, dest string) error {
+	if err := c.dial(); err != nil {
+		return fmt.Errorf("SFTP: unable to dial: %w", err)
+	}
+	defer c.close()
+
+	// Delete the file
+	if err := c.sftp.Remove(dest); err != nil {
+		if os.IsNotExist(err) {
+			return fmt.Errorf("SFTP: file does not exist: %w", err)
+		} else if os.IsPermission(err) {
+			return fmt.Errorf("SFTP: insufficient conditions to delete file: %w", err)
```

Comment:
I think "permissions" is clearer than "conditions".  Also, as you pointed out, dest could be a dir.

Suggested:
```
return fmt.Errorf("SFTP: insufficient permissions to delete: %w", err)
```

Reply:
good catch and good point